internal/controller/sync.go

```diff
@@ -368,41 +295,18 @@ func (r *Route) buildNextCR(ctx context.Context, route *routev1.Route, revision
 		privateKeySize, err = strconv.Atoi(privateKeySizeStr)
 		if err != nil {
 			r.eventRecorder.Event(route, corev1.EventTypeWarning, ReasonInvalidPrivateKeySize, "invalid private key size:"+privateKeySizeStr)
-			return nil, fmt.Errorf("invalid private key size, %s: %v", privateKeySizeStr, err)
+			return nil, fmt.Errorf("invalid private key size annotation %q on %s/%s", privateKeySizeStr, route.Namespace, route.Name)
 		}
 	}
 
-	var signatureAlgorithm x509.SignatureAlgorithm
-	var publicKeyAlgorithm x509.PublicKeyAlgorithm
-	switch privateKeyAlgorithm {
-	case string(cmapi.ECDSAKeyAlgorithm):
-		switch privateKeySize {
-		case 521:
-			signatureAlgorithm = x509.ECDSAWithSHA512
-		case 384:
-			signatureAlgorithm = x509.ECDSAWithSHA384
-		case 256:
-			signatureAlgorithm = x509.ECDSAWithSHA256
-		default:
-			signatureAlgorithm = x509.ECDSAWithSHA256
-		}
-		publicKeyAlgorithm = x509.ECDSA
-	case string(cmapi.RSAKeyAlgorithm):
-		switch {
-		case privateKeySize >= 4096:
-			signatureAlgorithm = x509.SHA512WithRSA
-		case privateKeySize >= 3072:
-			signatureAlgorithm = x509.SHA384WithRSA
-		case privateKeySize >= 2048:
-			signatureAlgorithm = x509.SHA256WithRSA
-		default:
-			signatureAlgorithm = x509.SHA256WithRSA
-		}
-		publicKeyAlgorithm = x509.RSA
+	var privateKeyRotationPolicy cmapi.PrivateKeyRotationPolicy
 
-	default:
-		r.eventRecorder.Event(route, corev1.EventTypeWarning, ReasonInvalidPrivateKeyAlgorithm, "invalid private key algorithm: "+privateKeyAlgorithm)
-		return nil, fmt.Errorf("invalid private key algorithm, %s", privateKeyAlgorithm)
+	if metav1.HasAnnotation(route.ObjectMeta, cmapi.PrivateKeyRotationPolicyAnnotationKey) {
+		// Don't validate the policy here because that would mean we'd need to update this codebase
+		// if cert-manager adds new values. Just rely on cert-manager validation when the cert is
+		// created
+		// This is brittle; ideally, cert-manager should expose a function for this
```

Comment:
nit(sorry, couldn't resist!!!): a sentence start with a capital letter and ends with a dot 😅

```suggestion
		// Don't validate the policy here because that would mean we'd need to update this codebase
		// if cert-manager adds new values. Just rely on cert-manager validation when the cert is
		// created. This is brittle; ideally, cert-manager should expose a function for 
		this.
```